lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -431,15 +432,25 @@ def current(self) -> Node:
                 temperature=current_node.get("attributes", {}).get("temp"),
             )
         except OpenSearchHttpError:
-            # we try to get the most accurate description of the node
-            conf_on_disk = self.config.load("opensearch.yml")
+            # we try to get the most accurate description of the node from the static config
+            if conf := self.config.load("opensearch.yml"):
+                roles = conf.get("node.roles")
+                app = App(id=conf.get("node.attr.app_id"))
```

Comment:
this will result in setting `None` values to those 2 attributes if the `opensearch.yml` exists but no value is set in those entries. (it's okay for the temperature).
  
A safer approach would be to use `conf[node.roles]` and catch the exception and use the `deployment_desc.config..` in the catch block or simply check on the existence of those 2 keys.

Reply:
Sure.
My assupmption was that --in case there IS a config-- these values MUST be set there.
(I'm still confused about what we should assume at which stage of the lifetime of the charm on these levels.)

Changing as requested.

Reply:
Done.